### PR TITLE
Fix highlight not working after HMD switch

### PR DESCRIPTION
**Pull request by SamGondelman**
https://highfidelity.manuscript.com/f/cases/17858/Selection-highlighting-stops-working-after-re-enter-HMD-several-times

Test plan:
- Enter HMD mode and laser on a grabbable entity. Notice that it highlights.
- Switch between HMD and desktop multiple times.  In HMD each time, the thing should highlight.
- Also check the context overlay (yellowish highlight outline when you right click or laser hover something from the marketplace)

### CODE REVIEW

File: libraries/render-utils/src/HighlightEffect.cpp [hunk 1]
```diff
@@ -58,32 +58,37 @@ void HighlightRessources::update(const gpu::FramebufferPointer& primaryFrameBuff
         if (!_colorFrameBuffer) {
             allocateColorBuffer(primaryFrameBuffer);
         }
+
+        // The primaryFrameBuffer render buffer can change
+        if (_colorFrameBuffer->getRenderBuffer(0) != primaryFrameBuffer->getRenderBuffer(0)) {
+            _colorFrameBuffer->setRenderBuffer(0, primaryFrameBuffer->getRenderBuffer(0));
+        }
```
Comment (SamGondelman) [inline libraries/render-utils/src/HighlightEffect.cpp]: This is the fix.  The render buffer can change from frame to frame, even if the size stays the same.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13735/HighFidelity-Beta-PR14169-61c5069.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13735/HighFidelity-Beta-PR14169-61c5069.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13735/HighFidelity-Beta-Interface-PR14169-61c5069.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13735/HighFidelity-Beta-PR14169-61c5069.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13735/HighFidelity-Beta-Interface-PR14169-61c5069.exe

File: libraries/render-utils/src/HighlightEffect.cpp [hunk 2]
```diff
@@ -314,7 +309,7 @@ void DrawHighlight::run(const render::RenderContextPointer& renderContext, const
 
                     batch.setViewportTransform(args->_viewport);
                     batch.setProjectionTransform(glm::mat4());
-                    batch.resetViewTransform();
+                    batch.setViewTransform(Transform());
```
Comment (SamGondelman) [inline libraries/render-utils/src/HighlightEffect.cpp]: @samcake does this matter?

Comment (samcake) [reply libraries/render-utils/src/HighlightEffect.cpp]: if you do resetViewTRanform you are doing the same and it s meant for that

Comment (SamGondelman) [reply libraries/render-utils/src/HighlightEffect.cpp]: Not exactly, resetViewTransform call setViewTransform with camera = false.  What does the camera parameter mean?

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13985/HighFidelity-Beta-PR14169-59fbfa4.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13985/HighFidelity-Beta-PR14169-59fbfa4.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13985/HighFidelity-Beta-Interface-PR14169-59fbfa4.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13985/HighFidelity-Beta-PR14169-59fbfa4.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13985/HighFidelity-Beta-Interface-PR14169-59fbfa4.exe

File: libraries/render-utils/src/HighlightEffect.cpp [hunk 3]
```diff
@@ -512,7 +509,7 @@ void DrawHighlightTask::build(JobModel& task, const render::Varying& inputs, ren
     const auto highlightSelectionNames = task.addJob<SelectionToHighlight>("SelectionToHighlight", sharedParameters);
 
     // Prepare for highlight group rendering.
-    const auto highlightRessources = task.addJob<PrepareDrawHighlight>("PrepareHighlight", primaryFramebuffer);
+    const auto HighlightResources = task.addJob<PrepareDrawHighlight>("PrepareHighlight", primaryFramebuffer);
```
Comment (AndrewMeadows) [inline libraries/render-utils/src/HighlightEffect.cpp]: Why this name change?  It doesn't appear to agree with our coding standard.

File: libraries/render-utils/src/HighlightEffect.h [hunk 4]
```diff
@@ -19,9 +19,9 @@
 #include "DeferredFramebuffer.h"
 #include "DeferredFrameTransform.h"
 
-class HighlightRessources {
+class HighlightResources {
```
Comment (AndrewMeadows) [inline libraries/render-utils/src/HighlightEffect.h]: With this class declaration the variable named `HighlightResources` above now seems like a Bad Idea.

Comment (SamGondelman) [reply libraries/render-utils/src/HighlightEffect.h]: typo, fixed

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14061/HighFidelity-Beta-PR14169-de4dff5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14061/HighFidelity-Beta-Interface-PR14169-de4dff5.exe

**Comment by SamGondelman** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14063/HighFidelity-Beta-PR14169-de4dff5.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/14063/HighFidelity-Beta-Interface-PR14169-de4dff5.dmg

**Comment by SamGondelman** [comment]
BUILD THIS PLEASE

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/14064/HighFidelity-Beta-PR14169-de4dff5.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14064/HighFidelity-Beta-PR14169-de4dff5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/14064/HighFidelity-Beta-Interface-PR14169-de4dff5.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
Highlight works as expected

**Pull request merged**